js/login.js

```diff
@@ -0,0 +1,60 @@
+document.addEventListener('DOMContentLoaded', function () {
+    const API_BASE_URL = 'https://api.noroff.dev';
+
+    async function loginUser(url, data) {
+        try {
+            const postData = {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify(data),
+            };
+            const response = await fetch(url, postData);
+            console.log(response);
+            const json = await response.json();
```

Comment:
You need to handle the error here. It is a good idea to check if the response returns valid data or not. As it is right now, your code breaks when response an invalid data. Here is an example
```
if (response.ok) {
                console.log(response);
                const json = await response.json();

                if (json.accessToken) {
                    // Store the access token in local storage
                    localStorage.setItem('accessToken', json.accessToken);
                    if (json.name) {
                        // Store the user's name in local storage
                        localStorage.setItem('userName', json.name);
                    }
                    console.log('Login successful');
                    // Redirect to profile page after successful login
                    window.location.href = 'profile/index.html';
                } else {
                    // Display the error message when login fails
                    const errorMessage = document.getElementById('login-error-message');
                    errorMessage.textContent = 'Login failed. Please check your credentials.';
                    errorMessage.classList.remove('d-none');
                    console.error('Login failed:', json.message);
                }
            } else {
                throw new Error("Response return an invalid data");
            }
```